pkg/logging/buffer.go

```diff
@@ -0,0 +1,97 @@
+package logging
+
+import (
+	"bytes"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// BufferLogger wraps a zap.SugaredLogger and implements the existing
+// LoggerInterface to capture logs for testing purposes.
+type BufferLogger struct {
+	Buffer  *bytes.Buffer
+	sugared *zap.SugaredLogger
+}
```

Comment:
```suggestion
type BufferLogger struct {
    *zap.SugaredLogger
	Buffer  *bytes.Buffer
}
```
Can we use such type definition and remove functions needed to satisfy interface?

Reply:
True, done.